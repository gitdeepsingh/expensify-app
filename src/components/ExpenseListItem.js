import React from 'react';
import moment from 'moment'
import numeral from 'numeral';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            <span>&#8377;</span>{numeral(amount).format('0,0.00')}
            -
            {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;
