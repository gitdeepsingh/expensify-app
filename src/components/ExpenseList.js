import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} /> 
                })
            )
        }
    </div>
);

const mapStateToProp = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
}

export default connect(mapStateToProp)(ExpenseList);
