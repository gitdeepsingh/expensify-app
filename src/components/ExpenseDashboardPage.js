import React from 'react';
import ExpenseList from './ExpenseList';
import ExpensesSummary from './ExpensesSummary';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />      
        <ExpenseListFilters />  
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;