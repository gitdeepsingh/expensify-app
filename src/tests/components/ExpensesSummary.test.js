import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={333} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={52452} />);
    expect(wrapper).toMatchSnapshot();
});
