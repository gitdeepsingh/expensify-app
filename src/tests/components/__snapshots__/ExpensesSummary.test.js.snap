// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
    <span>
       ₹
    </span>
    333.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling
    <span>
       ₹
    </span>
    52,452.00
  </h1>
</div>
`;
